source/core/ScriptDomain.cpp

```diff
@@ -96,69 +99,74 @@ namespace GTA
 			Log("[ERROR]", "Caught fatal unhandled exception:", Environment::NewLine, args->ExceptionObject->ToString());
 		}
 	}
-
-	ScriptDomain::ScriptDomain() : _appdomain(System::AppDomain::CurrentDomain), _executingThreadId(Thread::CurrentThread->ManagedThreadId), _runningScripts(gcnew List<Script ^>()), _taskQueue(gcnew Queue<IScriptTask ^>()), _pinnedStrings(gcnew List<IntPtr>()), _scriptTypes(gcnew List<Tuple<String ^, Type ^> ^>()), _recordKeyboardEvents(true), _keyboardState(gcnew array<bool>(256))
+	ScriptDomain::ScriptDomain() :
+		_appDomain(System::AppDomain::CurrentDomain),
+		_executingThreadId(Thread::CurrentThread->ManagedThreadId)
 	{
 		sCurrentDomain = this;
 
-		_appdomain->AssemblyResolve += gcnew ResolveEventHandler(&HandleResolve);
-		_appdomain->UnhandledException += gcnew UnhandledExceptionEventHandler(&HandleUnhandledException);
+		_appDomain->AssemblyResolve += gcnew ResolveEventHandler(&HandleResolve);
```

Comment:
@crosire Do you know if the current `AppDomain` ever changes? I have a feeling these lines of code are adding multiple copies of handlers to it, whereas there should only be one. The `HandleResolve` and `HandleUnhandledException` methods should be changed to `static` and a check put into place that checks to see if the current `AppDomain` has changed, and if so, add those static handlers to it.